### ISSUE TITLE
Import fails if the imported JSON has more language entries than the page currently has.

### ISSUE THREAD

**Issue by Ntsekees** [mkpoli/word-order#3]
I haven't yet tested extensively, but apparently, if you start with an illustration containing 4 languages (i.e. four lines), and you tries to import a JSON containing 5 languages (5 lines), it fails. However, if, before attempting importing, you start by adding a 5th language on the page, even a very simple thing like “language name = xx”, “sentence = zz” without doing any equivalency coloring at all, and then you import the same JSON that previously failed, then the import suddenly works seamlessly.

I have not investigated the code responsible for this issue, but I would suspect that somehow when importing a JSON with more languages than the current working data has, maybe the program fails to add enough “language space” to the data structure for harboring the additional language coming from the imported JSON? That's my best guess in any case.

—Ntsékees.

**Comment by mkpoli** [comment]
Yes. I also noticed the problem yesterday. It's apparently caused by the array `word_spans` isn't updated in time.

![image](https://user-images.githubusercontent.com/3502597/187200201-df96aa52-6fd1-4ad6-b35a-174150a36dfd.png)

**Comment by mkpoli** [comment]
This is a related problem but it only addresses removing not modifying in general, especially our case uses a 2d array for that
https://stackoverflow.com/questions/69848245/svelte-each-worked-twice-each-binding-error-cannot-set-properties-of-undefin

**Comment by mkpoli** [comment]
Fixed using a quite hacky solution, not very performant and elegant but at least fixes the problem.

Added a `loading` state, to delay svelte `bind:this={word_spans[i][j]}`, so it will only access `word_spans` after its size have been updated.

**Comment by Ntsekees** [comment]
Thanks!
